experimental/gcp-log/internal/storage/storage.go

```diff
@@ -298,6 +321,9 @@ func (c *Client) Sequence(ctx context.Context, leafhash []byte, leaf []byte) (ui
 		// writing to the same log.
 		w := bkt.Object(seqPath).If(gcs.Conditions{DoesNotExist: true}).NewWriter(ctx)
 		if _, err := w.Write(leaf); err != nil {
+			if c.otherCacheControl != "" {
+				w.ObjectAttrs.CacheControl = c.otherCacheControl
```

Comment:
This will only be set when the writes fail, is this intentional? Should they be set before the write?